### PR TITLE
[5]  implement parallel copy

### PR DESCRIPTION
**Pull request by jasoncoposky**

### CODE REVIEW

File: commands/cp/main.cpp [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+            int thread_count{4};
+
+            using rep_type = fs::object_time_type::duration::rep;
+
+            po::options_description desc{""};
+            desc.add_options()
+                ("logical_path", po::value<std::string>(), "logical path to collection or object to copy")
+                ("destination", po::value<std::string>(), "destination logical path for the copy")
+                ("unregister", po::bool_switch(&unregister), "unregister data instead of unlinking data")
+                ("no_trash", po::bool_switch(&no_trash), "do no move items to the trash can")
```
Comment (trel) [inline commands/cp/main.cpp]: no move -> not move

File: commands/rm/main.cpp [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+            bool progress_flag{false}, no_trash{false}, unregister{false};
+            int thread_count{4};
+
+            using rep_type = fs::object_time_type::duration::rep;
+
+            po::options_description desc{""};
+            desc.add_options()
+                ("logical_path", po::value<std::string>(), "logical path to collection or object to remove")
+                ("unregister", po::bool_switch(&unregister), "unregister data instead of unlinking data")
+                ("no_trash", po::bool_switch(&no_trash), "do no move items to the trash can")
```
Comment (trel) [inline commands/rm/main.cpp]: trash can - perhaps we just call the it the trash?

File: commands/rm/main.cpp [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+        auto help_text() const noexcept -> std::string_view override
+        {
+            auto help =R"(
+Given a fully qualfied or relative path, remove the collection or object
+
+irods rm [options] fully_qualified_logical_path
+
+      --unregister        : unregister data instead of unlinking data
+      --no_trash          : do no move items to the trash can
```
Comment (trel) [inline commands/rm/main.cpp]: "not move"    "the trash"   no can

**Review by trel — COMMENTED** [review]
got tests?

File: commands/rm/main.cpp (hunk 2 above)
Comment (jasoncoposky) [reply commands/rm/main.cpp]: this reflects what is in file system, we can take a uniform pass over the whole thing if necessary.

File: commands/cp/main.cpp (hunk 1 above)
Comment (jasoncoposky) [reply commands/cp/main.cpp]: resolved, cruft.

File: commands/rm/main.cpp (hunk 3 above)
Comment (jasoncoposky) [reply commands/rm/main.cpp]: resolved, cruft.

File: commands/cp/main.cpp [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+        auto description() const noexcept -> std::string_view override
+        {
+            return "Removes collections or data objects";
+        }
+
+        auto help_text() const noexcept -> std::string_view override
+        {
+            auto help =R"(
+Given a fully qualfied or relative path, remove the collection or object
```
Comment (trel) [inline commands/cp/main.cpp]: qualified

**Comment by jasoncoposky** [comment]
also, there do not appear to be any tests associated with this repo.

**Comment by trel** [comment]
discush!    yes, part of the conversation about 'what does a fully client side test harness/environment look like?'.

**Comment by trel** [comment]
please # if it's working at the bench

**Pull request merged**